packages/mux-player-react/src/lazy.tsx

```diff
@@ -0,0 +1,93 @@
+import React, { Suspense, useEffect, useState } from 'react';
+
+import useIsBrowser from './useIsBrowser';
+import useIsIntersecting from './useIsIntersecting';
+
+import type { MuxPlayerProps, MuxPlayerRefAttributes } from './index';
+
+const MuxPlayerIndex = React.lazy(() => import('./index'));
+
+const Overlay = () => (
+  <div
+    data-mux-player-react-lazy-overlay
+    style={{
+      position: 'absolute',
+      inset: 0,
+      backgroundColor: 'var(--controls-backdrop-color, rgba(0, 0, 0, 0.6))',
+    }}
+  />
+);
+
+interface WrapperProps {
+  onIntersection?: () => void;
+  placeholder?: string;
+  style?: React.CSSProperties;
+  className?: string;
+  children?: React.ReactNode;
+}
+const Wrapper = ({ onIntersection, placeholder, style, className, children }: WrapperProps) => {
+  const intersectionRef = React.useRef<HTMLDivElement>(null);
+  const isIntersecting = useIsIntersecting(intersectionRef);
+
+  useEffect(() => {
+    if (isIntersecting && onIntersection) {
+      onIntersection();
+    }
+  }, [isIntersecting, onIntersection]);
+
+  return (
+    <div
+      ref={intersectionRef}
+      data-mux-player-react-lazy
+      style={{
+        // default mux-player styling
+        aspectRatio: '16/9',
+        display: 'block',
+        width: '100%',
```

Comment:
One downside to this approach? This wrapper doesn't automagically inherit mux player's styling. If we make a styling change to mux player, we'll have to make it here, too.